### PR TITLE
Release 3.4.2 bugfix release

### PR DESCRIPTION
**Pull request by PhilipVinc**

### CODE REVIEW

**Comment by femtobit** [comment]
@PhilipVinc I'd like to add the correction we discussed to the changelog, while we're add it. Is this okay (see my commit)?

**Comment by PhilipVinc** [comment]
yes it's good for me 
(I still want to fix a few more of those dtype things before releasing)

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,21 +13,35 @@
...
+
+### Internal Changes
+* Several deprecation warnings related to `jax.experimental.loops` being deprecated have been resolved by changing those calls to `jax.lax.fori_loop`. Jax should feel more tranquillo now. [#1172](https://github.com/netket/netket/pull/1172)
 
 ### Bug Fixes
 * Several _type promotion_ bugs that would end up promoting single-precision models to double-precision have been squashed. Those involved `nk.operator.Ising` and `nk.operator.BoseHubbard`[#1180](https://github.com/netket/netket/pull/1180), `nkx.TDVP` [#1186](https://github.com/netket/netket/pull/1186) and continuous-space samplers and operators [#1187](https://github.com/netket/netket/pull/1187).
+* `nk.operator.Ising`, `nk.operator.BoseHubbard` and `nk.operator.LocalLiouvillian` now return connected samples with the same precision (`dtype`) as the input samples. This allows to preserve low precision along the computation when using those operators.[#1180](https://github.com/netket/netket/pull/1180)
+* `nkx.TDVP` now updates the expectation value displaied in the progress bar at every time-step. [#1182](https://github.com/netket/netket/pull/1182)
```
Comment (femtobit) [inline CHANGELOG.md:30]:
```suggestion
* `nkx.TDVP` now updates the expectation value displayed in the progress bar at every time step. [#1182](https://github.com/netket/netket/pull/1182)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,21 +13,35 @@
...
 * Fixed bug [#1192](https://github.com/netket/netket/pull/1192) that affected most operators (`nk.operator.LocalOperator`) constructed on non-homogeneous hilbert spaces. This bug was first introduced in version 3.3.4 and affects all subsequent versions until [NEXT BUGFIX RELEASE]. [#1193](https://github.com/netket/netket/pull/1193)
```
Comment (femtobit) [inline CHANGELOG.md:31]:
```suggestion
* Fixed bug [#1192](https://github.com/netket/netket/pull/1192) that affected most operators (`nk.operator.LocalOperator`) constructed on non-homogeneous hilbert spaces. This bug was first introduced in version 3.3.4 and affects all subsequent versions until 3.4.2. [#1193](https://github.com/netket/netket/pull/1193)
```

**Review by femtobit — CHANGES_REQUESTED** [review]
Looks good to me.

**Review by femtobit — APPROVED** [review]

**Review by gcarleo — APPROVED** [review]

**Pull request merged**